client/state/login/actions.js

```diff
@@ -572,3 +575,39 @@ export const logoutUser = redirectTo => ( dispatch, getState ) => {
 			return Promise.reject( error );
 		} );
 };
+
+/**
+ * Retrieves the type of authentication of the account of the specified user.
+ *
+ * @param {String} usernameOrEmail - id of the user
+ * @return {Function} a promise that will resolve once the authentication account type has been retrieved
+ */
+export const getAuthAccountType = ( usernameOrEmail ) => dispatch => {
+	dispatch( {
+		type: LOGIN_AUTH_ACCOUNT_TYPE_REQUEST,
+	} );
+
+	if ( usernameOrEmail === '' ) {
+		const error = {
```

Comment:
Why not an instance of `Error` ?

Reply:
Here I'm just matching the data structure returned by `getErrorFromHTTPError()`:

https://github.com/Automattic/wp-calypso/blob/47a486525668a2637236c2329d9b8e4568a086bd/client/state/login/actions.js#L80-L84